src/modules/report/yaml.py

```diff
@@ -2,41 +2,51 @@
 
 from ruamel.yaml import YAML
 
-from collector import services, vulnerabilities
+from collector import services, vulnerabilities, services_lock, vulnerabilities_lock
 
 
 class YAMLReporter(object):
     def get_report(self):
         yaml = YAML()
+        vulnerabilities_lock.acquire()
```

Comment:
Do you need this lock here? There's no obvious access to the variable it protects. (Adding locks where they aren't necessary increases the likelihood that there will be a deadlock introduced.)